nats.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"time"
+
+	nats "github.com/nats-io/nats-server/server"
+)
+
+var defaultNatsOptions = &nats.Options{
+	Host:   "0.0.0.0",
+	Port:   4222,
+	NoLog:  true,
+	NoSigs: true,
+	Cluster: nats.ClusterOpts{
+		Host: "0.0.0.0",
+		Port: 4248,
+	},
+}
+
+func (s *server) runNatsServer(opts *nats.Options) {
+	s.nats = nats.New(opts)
+	go s.nats.Start()
+	// Wait for accept loop(s) to be started
+	if !s.nats.ReadyForConnections(10 * time.Second) {
+		panic("Unable to start NATS Server in Go Routine")
```

Comment:
Can we throw error instead of panic?

Reply:
We need to force quit if this error occurs otherwise the other modules won't work. Can use a log.Fatal instead?